packages/pyright-internal/src/tests/fourslash/completions.importInterimFile.fourslash.ts

```diff
@@ -0,0 +1,52 @@
+/// <reference path="fourslash.ts" />
+
+// @filename: test1.py
+//// import al[|/*marker1*/|]
+
+// @filename: altair/__init__.py
+// @library: true
+//// """module docstring"""
+////
+//// __all__ = [ "selection_interval" ]
+////
+//// from .vegalite import (
+////     selection,
+////     selection_interval
+//// )
+
+// @filename: altair/vegalite/__init__.py
+// @library: true
+//// def selection(): pass
+//// def selection_interval(): pass
+
+// @filename: altair/py.typed
+// @library: true
+//// # has to contain something for file to be written
+
+{
+    // Force interim file to be created
+    // @ts-ignore
+    await helper.verifyCompletion('included', 'markdown', {
```

Comment:
why interim file is created here?

Reply:
originally `interim` file meant to indicate a file that is not `rooted` by any other file so, it would go away when `removeUnncessaryFile` is called.

by `rooted`, it means a file is used (imported) by user files (isTracked) or open files transitively.

any file that are not imported by rooted files transitively such as `pytest fixtures in `conftest.py` that are not imported by any thing` or `module in string literal that are not imported by anything` and etc are created as `interim` file.

everything involved here seems explicitly imported? so not sure what would case interim file to be created?

it seems not using any shadow file as well which could also cause `interim` file to be created.

Reply:
Here's the callstack where it gets hit in this scenario. This occurs before the user has completely typed in the module name `altair` (test.py contains `import al`), so at that point it's not explicitly imported. When manually reproing the issue, I've noticed that it doesn't repro if you type slowly after the interim file is created, which I believe is because you're allowing `removeUnneccessaryFiles` to get called.

```text
Program._createInterimFileInfo (c:\src\pyright\packages\pyright-internal\src\analyzer\program.ts:1578)
Program.addInterimFile (c:\src\pyright\packages\pyright-internal\src\analyzer\program.ts:351)
<anonymous -- this is SourceMapper._boundSourceGetter> (c:\src\pyright\packages\pyright-internal\src\analyzer\program.ts:1212)
SourceMapper.findModules (c:\src\pyright\packages\pyright-internal\src\analyzer\sourceMapper.ts:65)
getModuleDocStringFromPaths (c:\src\pyright\packages\pyright-internal\src\analyzer\typeDocStringUtils.ts:203)
CompletionProvider.resolveCompletionItem (c:\src\pyright\packages\pyright-internal\src\languageService\completionProvider.ts:350)
resolveCompletionItem (c:\src\pyright\packages\pyright-internal\src\tests\harness\fourslash\testState.ts:1522)
```